tutorials/strings/manachers-algorithm.md

```diff
@@ -9,4 +9,174 @@ keywords:
   - algorithm
 ---
 
-Coming Soon
\ No newline at end of file
+
+# Manacher's Algorithm
+
+There are many ways to find the `longest palindromic substring`. One can find it in O(N³) and also in O(N²) time complexity. But using `Manacher's Algorithm` we do it in `O(N) time`. This algorithm was discovered by `Glenn K. Manacher` in 1975.
+
+Given a string, find the longest substring which is palindrome. 
+
+* if the given string is “abaaba”, the output should be “abaaba”
+* if the given string is “abababa”, the output should be “abababa”
+* if the given string is “abcbabcbabcba”, the output should be “abcbabcbabcba”
+
+* ## Brute force Approach O(N³) time
+
+```C++
+void longestPalSubstring(string str){
+    // Getting length of the input string
+    int n = str.size();
+
+    // All substrings of length 1 are palindromes
+    int maxLength = 1;
+    
+    int start = 0;
+
+    // Checking all the substrings
+    for (int i = 0; i < n; i++){
+        for (int j = i; j < n; j++){
+            int flag = 1;
+
+            // Checking for a palindromic subtring
+            for (int k = 0; k < (j - i + 1) / 2; k++)
+                if (str[i + k] != str[j - k])
+                    flag = 0;
+
+            // If substring is palindromic
+            if (flag && (j - i + 1) > maxLength){
+                start = i;
+                maxLength = j - i + 1;
+            }
+        }
+    }
+
+    // Printing the longest Palindromic substring
+    cout << "The Longest Palindromic Substring is: ";
+    for (int i = start; i <= start+maxLength-1; i++)
+        cout << str[i];
+}
+```
+### Input: 
+daabddfddbegtd
+### Output: 
+The Longest Palindromic Substring is: bddfddb
+### Time complexity O(N³)
+O(N³)
+In the brute force approach, three nested loops are used to find the longest palindromic substring, so the time complexity will be O(N³).
+### Space complexity O(1)
+No extra space is needed in this approach, so the space complexity will be O(1).
+
+* ## Manacher's Algorithm
+This algorithm is faster than the brute force approach, as it exploits the idea of a palindrome happening inside another palindrome.
+
```

Comment:
The content starting from this line is copied from https://www.scaler.com/topics/data-structures/manachers-algorithm/. What's the point of copying other content exactly?